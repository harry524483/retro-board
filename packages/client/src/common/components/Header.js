import { Label, Header as SemanticHeader, Divider } from 'semantic-ui-react';

const Header = ({ title, children, label }) => {
  return (
    <>
      <SemanticHeader as="h2">
        {title}
        {label && <Label circular>{label}</Label>}
      </SemanticHeader>
      <Divider horizontal>
        <SemanticHeader as="h4" color="blue">
          {children}
        </SemanticHeader>
      </Divider>
    </>
  );
};

export default Header;
