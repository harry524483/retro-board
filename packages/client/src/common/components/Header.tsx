import { FC } from 'react';
import { Label, Header as SemanticHeader, Divider } from 'semantic-ui-react';

export type Props = { title: string; label: string };

const Header: FC<Props> = ({ title, children, label }): JSX.Element => {
  return (
    <>
      <SemanticHeader as="h2" data-testid="main-header">
        {title}
        {label && (
          <Label circular data-testid="label">
            {label}
          </Label>
        )}
      </SemanticHeader>
      <Divider horizontal data-testid="divider">
        <SemanticHeader as="h4" color="blue">
          {children}
        </SemanticHeader>
      </Divider>
    </>
  );
};

export default Header;
