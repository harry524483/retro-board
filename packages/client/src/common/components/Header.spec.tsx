import { render } from '@testing-library/react';

import Header, { Props } from './Header';

it('renders without error', () => {
  // Arrange
  const children = 'I am children';
  const props: Props = { title: 'foo', label: 'bar' };

  // Act
  const { asFragment } = render(<Header {...props}>{children}</Header>);

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
