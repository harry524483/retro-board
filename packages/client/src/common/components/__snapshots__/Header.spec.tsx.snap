// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without error 1`] = `
<DocumentFragment>
  <h2
    class="ui header"
  >
    foo
    <div
      class="ui circular label"
    >
      bar
    </div>
  </h2>
  <div
    class="ui horizontal divider"
  >
    <h4
      class="ui blue header"
    >
      I am children
    </h4>
  </div>
</DocumentFragment>
`;
